### PR TITLE
Run-checker converted to GitHub Actions

### PR DESCRIPTION
**Pull request by paulidale**
- [ ] documentation is added or updated
- [x] tests are added or updated

### CODE REVIEW

**Review by slontis — COMMENTED** [review]
I assume it doesnt like spaces in an array of comma seperate entries..

**Review by slontis — COMMENTED** [review]
#Run once a week
Use schedule: -cron if you want to run less frequently..

on:
  schedule:
    - cron:  '0 0 * * SAT'

**Comment by paulidale** [comment]
I need the run on `pull_request` to figure out how to get this working.
I'll switch to different timing later.

**Comment by slontis** [comment]
yep - had the same problem

File: .github/workflows/run-checker.yml [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+          no-egd,
+          no-engine,
+          no-engine no-shared,
+          no-err,
+          no-external-tests,
+          no-filenames,
+          enable-fips,
+          enable-fips enable-acvp-tests,
+          enable-fips no-cmac,
+          enable-fips no-cmac,
```
Comment (slontis) [inline .github/workflows/run-checker.yml:68]: entry repeated here for cmac?

**Review by slontis — COMMENTED** [review]
I realize this is trying to cover everything.. But is there really any point in testing 'no-' options for options that are already disabled by default??

Disabled features:
    asan                [default]        OPENSSL_NO_ASAN
    buildtest-c++       [default]        
    crypto-mdebug       [default]        OPENSSL_NO_CRYPTO_MDEBUG
    devcryptoeng        [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128 [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                 [default]        OPENSSL_NO_EGD
    external-tests      [default]        OPENSSL_NO_EXTERNAL_TESTS
    fips                [default]        
    fips-securitychecks [cascade]        OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl            [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer      [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                [default]        OPENSSL_NO_KTLS
    md2                 [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                [default]        OPENSSL_NO_MSAN
    rc5                 [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                [default]        OPENSSL_NO_SCTP
    ssl-trace           [default]        OPENSSL_NO_SSL_TRACE
    trace               [default]        OPENSSL_NO_TRACE
    ubsan               [default]        OPENSSL_NO_UBSAN
    unit-test           [default]        OPENSSL_NO_UNIT_TEST
    uplink              [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers    [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]        
    zlib-dynamic        [default]        
    ssl3                [default]        OPENSSL_NO_SSL3
    ssl3-method         [default]        OPENSSL_NO_SSL3_METHOD

**Comment by paulidale** [comment]
Probably should test the enabled- versions.  I just pulled that run-checker did.  I'm not surprised it's out of date.

File: .github/workflows/run-checker.yml (hunk 1 above)
Comment (paulidale) [reply .github/workflows/run-checker.yml:68]: My stuff up :(

**Comment by slontis** [comment]
Note the failure for  no-ec2m is discussed in https://github.com/openssl/openssl/issues/15170

**Comment by slontis** [comment]
@paulidale - I know what the issue related to the no-cached-fetch is in the dsa_test.. PR will be coming soon for that.
Another  export/import bug related to an extra flag that I tacked on to something in the ffc_params :(

**Comment by slontis** [comment]
I will leave the other one related to memleak to you..

**Comment by paulidale** [comment]
Both are recent regressions :(

**Comment by paulidale** [comment]
Memleak fix in #15195

**Comment by paulidale** [comment]
I agree, the list here is way too large.

The PR is adding 18 CI jobs on push.  I think that's a manageable number of additional tests.  We can always tweak things later.

**Comment by t8m** [comment]
> Both are recent regressions :(

The memleak is a regression from dropping the -d from runchecker jobs. :grin:

**Comment by paulidale** [comment]
A better fix might be to drop the asan and ubsan from that particular build.

**Comment by paulidale** [comment]
Moving this out of WIP.

There are two daily test failures:
- the `ec2m` which is a decoder problem and
- the `dsa_test` which @slontis is already onto.

File: .github/workflows/run-checker-daily.yml [hunk 2]
```diff
@@ -24,7 +24,7 @@ jobs:
           no-blake2,
           no-buildtest-c++,
           no-bulk,
-          no-cached-fetch enable-asan enable-ubsan, # Very slow
+          no-cached-fetch
```
Comment (t8m) [inline .github/workflows/run-checker-daily.yml:27]: Missing comma

Comment (paulidale) [reply .github/workflows/run-checker-daily.yml]: Great catch.

Fixed.

**Review by t8m — APPROVED** [review]
Let's approve, we can always modify the jobs later.

**Comment by paulidale** [comment]
Made a wording fix in a comment _daily_ -> _day_.
Will need re-approval but I don't think the timer needs resetting.

**Review by slontis — APPROVED** [review]

**Comment by paulidale** [comment]
Merged to master.  Thanks for all of the input.

**Pull request closed without merging**